langserve/packages.py

```diff
@@ -0,0 +1,87 @@
+import importlib
+import logging
+from pathlib import Path
+from typing import Generator, TypedDict, Union
+
+from fastapi import APIRouter, FastAPI
+from tomli import load
+
+from langserve.server import add_routes
+
+
+class LangServeExport(TypedDict):
+    module: str
+    attr: str
+    package_name: str
+
+
+def get_langserve_export(filepath: Path) -> LangServeExport:
+    with open(filepath, "rb") as f:
+        data = load(f)
+    try:
+        module = data["tool"]["langserve"]["export_module"]
+        attr = data["tool"]["langserve"]["export_attr"]
+        package_name = data["tool"]["poetry"]["name"]
+    except KeyError as e:
+        raise KeyError("Invalid LangServe PyProject.toml") from e
+    return LangServeExport(module=module, attr=attr, package_name=package_name)
+
+
+exclude_paths = set(["__pycache__", ".venv", ".git", ".github"])
+
+
+def _include_path(path: Path) -> bool:
+    """
+    Skip paths that are in exclude_paths or start with an underscore.
+    """
+    for part in path.parts:
+        if part in exclude_paths:
+            return False
+        if part.startswith("_"):
+            return False
+    return True
+
+
+def list_packages(path: str = "../packages") -> Generator[Path, None, None]:
+    # traverse path for routes to host (any directory holding a pyproject.toml file)
+    package_root = Path(path)
+    for pyproject_path in package_root.glob("**/pyproject.toml"):
```

Comment:
Are all packages with pyproject.toml usable or do they need `langserve` tool definition?

Reply:
They need the `langserve` tool definition, and changed the logic to just warn if that's not the case